src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs

```diff
@@ -4867,6 +4860,8 @@ internal void CleanupWindowsPowerShellCompatResources()
             }
         }
 
+        internal virtual IList<PSModuleInfo> ImportModulesUsingWinCompat(IEnumerable<string> moduleNames, IEnumerable<ModuleSpecification> moduleFullyQualifiedNames, ImportModuleOptions importModuleOptions) {throw new System.NotImplementedException();}
```

Comment:
Is there a reason to make this virtual with only one override implementation?

Reply:
The situation is like this... the compat-import-function has to be visible to/called from both parent class (`ModuleCmdletBase.LoadModuleManifest`) and child class (`ImportModuleCommand.ProcessRecord`); at the same time it uses function of the child class (`ImportModuleCommand.ImportModule_RemotelyViaPsrpSession`), so it can not be defined in the parent class. Having a virtual function in the parent with the override in the child class is probably the easiest way to achieve the required behavior.
Code-design-wise this is definitely not the target scenario for using virtual functions, but it gets the job done, considering that module code (`ImportModuleCommand` / `ModuleCmdletBase` classes) is already so fused (functionality of child class is defined in the parent) that you can not make it worse.

Reply:
If it needs to be called in ModuleCmdletBase, it's probably best to just put the implementation there; that's the point of ImportModuleCommand overriding ModuleCmdletBase, so that it can inherit the methods.

Reply:
If it is placed in `ModuleCmdletBase`, then it won't see `ImportModuleCommand.ImportModule_RemotelyViaPsrpSession` that it needs to call. :)

Reply:
🤦‍♂ Well hopefully we can address the architectural issues here when we look into the minimal PowerShell effort

Reply:
We should definitely pay attention to this during minimal PowerShell effort. If the import code stayed in `ImportModuleCommand` and didn't over time partially migrate to the parent `ModuleCmdletBase`, then there would be no need to dance around this mess with crazy workarounds like this one.